spec/feature/users_spec.rb

```diff
@@ -0,0 +1,112 @@
+require "rails_helper"
+require "spec_helper"
+
+feature "Registration" do
+  scenario "User can became registered" do
+    visit root_path
+    click_link "Register"
+    fill_in "Email", with: "new_user@mail.ru"
+    fill_in "Password", with: "secret"
+    fill_in "Password confirmation", with: "secret"
+    click_button "Create User"
+
+    expect(page).to have_content "User was successfully created"
+    expect(page).to have_content "Logout"
+
+  end
+end
+
+feature "Login and logout" do
+  scenario "User can login with right email and password" do
+    create(:user)
+    login
+    expect(page).to have_content "Login successful"
+  end
+
+  scenario "User can logout" do
+    create(:user)
+    login
+    click_link "Logout"
+    expect(page).to have_content "Logged out!"
+  end
+
+  scenario "User can't login with wrong email" do
+    create :user
+    visit root_path
+    fill_in "email", with: "sfasfd89899899"
+    fill_in "password", with: "secret"
+    click_button "Login" 
+    expect(page).to have_content "Login failed"
+  end
+
+  scenario "user can't login with wrong pass" do
+    create :user
+    visit root_path
+    fill_in "email", with: "mail@mail.ru"
+    fill_in "password", with: "78738778738817"
+    click_button "Login"
+    expect(page).to have_content "Login failed"
+  end
+
+  scenario "user can't login with wrong pass and wrong email" do
+    create :user
+    visit root_path
+    fill_in "email", with: "123123123123"
+    fill_in "password", with: "78738778738817"
+    click_button "Login"
+    expect(page).to have_content "Login failed"
+  end
+end
+
+feature "User see only his cards" do
+  scenario "one user can't see card of another" do
+  first_user = create(:user, email: "first@mail.ru", password: "111", password_confirmation: "111")
+  create(:card, original_text: "first", translated_text: "первый", user: first_user)
+  second_user = create(:user, email: "second@mail.ru", password: "111", password_confirmation: "111")
+  create(:card, original_text: "second", translated_text: "второй", user: second_user)
+
+  visit root_path
+  fill_in "email", with: "first@mail.ru"
+  fill_in "password", with: "111"
+  click_button "Login"
+  click_link   "Карточки"
```

Comment:
Unnecessary spacing detected.<br>Put one space between the method name and the first argument.